sql/src/main/java/io/crate/expression/tablefunctions/GenerateSeries.java

```diff
@@ -0,0 +1,197 @@
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.expression.tablefunctions;
+
+import io.crate.action.sql.SessionContext;
+import io.crate.analyze.WhereClause;
+import io.crate.data.Bucket;
+import io.crate.data.Input;
+import io.crate.data.Row;
+import io.crate.data.RowN;
+import io.crate.metadata.BaseFunctionResolver;
+import io.crate.metadata.ColumnIdent;
+import io.crate.metadata.FunctionIdent;
+import io.crate.metadata.FunctionImplementation;
+import io.crate.metadata.FunctionInfo;
+import io.crate.metadata.Reference;
+import io.crate.metadata.ReferenceIdent;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.Routing;
+import io.crate.metadata.RoutingProvider;
+import io.crate.metadata.RowGranularity;
+import io.crate.metadata.functions.params.FuncParams;
+import io.crate.metadata.functions.params.Param;
+import io.crate.metadata.table.StaticTableInfo;
+import io.crate.metadata.table.TableInfo;
+import io.crate.metadata.tablefunctions.TableFunctionImplementation;
+import io.crate.types.DataType;
+import io.crate.types.DataTypes;
+import org.elasticsearch.cluster.ClusterState;
+
+import javax.annotation.Nonnull;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.function.BinaryOperator;
+
+/**
+ * <pre>
+ * {@code
+ *      generate_series :: a -> a -> table a
+ *      generate_series(start, stop)
+ *
+ *      generate_series :: a -> a -> a -> table a
+ *      generate_series(start, stop, step)
+ *
+ *      where: a = Integer or Long
+ * }
+ * </pre>
+ */
+public final class GenerateSeries<T extends Number> extends TableFunctionImplementation<T> {
+
+    public static final String NAME = "generate_series";
+    private static final RelationName RELATION_NAME = new RelationName("", NAME);
+    private final FunctionInfo info;
+    private final T defaultStep;
+    private final BinaryOperator<T> minus;
+    private final BinaryOperator<T> plus;
+    private final BinaryOperator<T> divide;
+    private final Comparator<T> comparator;
+
+    public static void register(TableFunctionModule module) {
+        Param longOrInt = Param.of(DataTypes.LONG, DataTypes.INTEGER);
+        FuncParams.Builder paramsBuilder = FuncParams
+            .builder(longOrInt, longOrInt)
+            .withVarArgs(longOrInt)
+            .limitVarArgOccurrences(1);
+        module.register(NAME, new BaseFunctionResolver(paramsBuilder.build()) {
+            @Override
+            public FunctionImplementation getForTypes(List<DataType> types) throws IllegalArgumentException {
+                DataType dataType = types.get(0);
+                if (dataType.equals(DataTypes.INTEGER)) {
```

Comment:
I wonder if we should drop the int specific handling and just go with long. We've implicit `long -> int` casts anyway. And the literals from statements are also long. That would allow us to drop this type generification

Thoughts?

Reply:
I wonder as well without any strong opinion, I'm not really sure if it matters in practice to only support/return longs.